parlai/nn/lr_scheduler.py

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""
+Code for LR Schedulers.
+
+See ParlAILRScheduler (super class) and subclasses for detailed documentation
+"""
+
+from abc import abstractmethod
+from torch import optim
+import numpy as np
+
+from parlai.utils.exceptions import StopTrainException
+from parlai.utils.misc import warn_once
+
+
+class ParlAILRScheduler(object):
+    """
+    Class for LR Schedulers Subclasses must implement abstract methods train_step() and
+    valid_step() Schedulers can be initialized with lr_scheduler_factory()
+    """
+
+    def __init__(self, optimizer, states, hard_reset, warmup_updates):
+        """
+        Initialize warmup scheduler.
```

Comment:
It's more than that, no?

You should also say that this isn't meant to be initialized directly, but rather use the factory.